### PR TITLE
cleaned up examples; added automagic open on startup

### PR DESCRIPTION
**Pull request by neilff**
This PR removes all usage of `R.element` and replaces it with the corresponding JSX. Along with this, I'm trying to improve the developer experience while working on Formatic. 

Opening this as someone new to the project can be a bit intimidating with all of the code jammed into the examples file. This takes the examples out of the `examples/index.js` and moves them into individual files.

As well, it adds a _super_ simple debugger for understanding what the form state looks like:

image

### CODE REVIEW

**Comment by neilff** [comment]
[JSX-ify Formatic](https://trello.com/c/NJNgFMgV/56-jsx-ify-formatic)

**Comment by jdeal** [comment]
@neilff Guessing this is a work in progress, but I tried to run this on my machine, and I get:

```bash
➜  formatic git:(improve-dx) npm start

> formatic@0.3.71 start /Users/justin/git/formatic
> node server.js

[ 'webpack-dev-server/client?http://localhost:3001',
  'webpack/hot/only-dev-server',
  './demo/index.js' ]
module.js:442
    throw err;
    ^

Error: Cannot find module 'node-libs-browser'
    at Function.Module._resolveFilename (module.js:440:15)
    at Function.Module._load (module.js:388:25)
    at Module.require (module.js:468:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/justin/git/formatic/node_modules/webpack/lib/node/NodeSourcePlugin.js:7:23)
    at Module._compile (module.js:541:32)
    at Object.Module._extensions..js (module.js:550:10)
    at Module.load (module.js:458:32)
    at tryModuleLoad (module.js:417:12)
    at Function.Module._load (module.js:409:3)

npm ERR! Darwin 16.4.0
npm ERR! argv "/Users/justin/.nvm/versions/node/v6.2.2/bin/node" "/Users/justin/.nvm/versions/node/v6.2.2/bin/npm" "start"
npm ERR! node v6.2.2
npm ERR! npm  v3.9.5
npm ERR! code ELIFECYCLE
npm ERR! formatic@0.3.71 start: `node server.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the formatic@0.3.71 start script 'node server.js'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the formatic package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node server.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs formatic
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls formatic
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/justin/git/formatic/npm-debug.log
```

**Comment by neilff** [comment]
@jdeal that issue should now be resolved, I also updated the eslint to match zapier-core and fixed the issues that came up as a result of doing that

File: demo/examples/json.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const fields = [
+  {
+    label: 'The Blob',
+    type: 'json',
+    key: 'blob',
+    readOnly: true
```
Comment (jdeal) [inline demo/examples/json.js]: Any reason this is `readOnly`? Thinking it shouldn't be.

**Comment by neilff** [comment]
Ohh, I didn't even realize those were aliases 🙃 , that makes _a lot_ more sense now. 

At a certain point I just put my head down and built out those example files without really thinking too much about them. I think your ideas there make the most sense: have the canonical name as the original / listing on the example page, and then mention the available aliases under each example.

**Comment by neilff** [comment]
@jdeal I've moved aliases to be under their base entry, I added notes where applicable. Also dropped that read-only on the JSON entry

**Comment by jdeal** [comment]
@neilff Sorry to nitpick since this such an improvement, but any reason not to just have an `aliases` array instead of using `notes`? Seems like you could easily change the rendering of those aliases that way. With `notes`, it's not very DRY.

**Comment by neilff** [comment]
@jdeal added the changes for `aliases`, please take a 👀  when you have a sec!

**Comment by jdeal** [comment]
@neilff Looks better to me. I went ahead and added a couple more tweaks though. The aliases list made it seemed like the main type was missing, so I added that:



^ Also, I added that "Example JSON" field which if you expand it, you can see and edit the JSON for the example field/fields:

**Comment by neilff** [comment]
@jdeal cool, those changes make sense to me 👍

**Pull request merged**